python/hail/table.py

```diff
@@ -336,27 +315,29 @@ def __getitem__(self, item):
             exprs = item if isinstance(item, tuple) else (item,)
             return self.index(*exprs)
 
-    def __iter__(self):
-        raise TypeError("'Table' object is not iterable")
-
     @property
-    def schema(self):
-        if self._schema is None:
-            self._schema = HailType._from_java(self._jt.signature())
-            assert (isinstance(self._schema, tstruct))
-        return self._schema
+    def key(self):
+        """Row key struct.
 
-    @property
-    def fields(self):
-        if self._field_names is None:
-            self._field_names = list(self._jt.fieldNames())
-        return self._field_names
+        Examples
+        --------
+
+        Get the row key field names:
+
+        .. doctest::
+
+            >>> list(table1.key)
+            ['ID']
+
```

Comment:
Since you deleted `n_fields`, can you add examples that return the number of fields for `key`, `row`, and `globals`?